gettsim/benefits/elterngeld.py

```diff
@@ -0,0 +1,215 @@
+        person,
+        e_st_params,
+        soli_st_params,
+        beit_bem_grenz=soz_vers_beitr_params[f"rvmaxek{westost}"],
+        werbungs_pausch=e_st_abzuege_params["werbung"],
+        soz_vers_pausch=params["soz_vers_pausch"],
+    )
+
+    current_net_wage = calc_net_wage(person)
+
+    return net_wage_last_year - current_net_wage
+
+
+def calc_elterngeld_percentage(considered_wage, params):
+    """ This function calculates the percentage share of net income, which is
+    reinbursed in the time of recieving elterngeld.
+
+    According to § 2 (2) BEEG the percentage increases below elg_st_1 and decreases
+    above elg_st_2.
+    """
+    if considered_wage < params["elg_st_1"]:
+        wag_diff = params["elg_st_1"] - considered_wage
+
+        number_steps = wag_diff / params["elg_perc_correct_step"]
+        percentage = params["elgfaktor"] + number_steps * params["elg_perc_correct"]
+
+    elif considered_wage > params["elg_st_2"]:
+        wag_diff = considered_wage - params["elg_st_2"]
+
+        number_steps = wag_diff / params["elg_perc_correct_step"]
+        percentage = max(
+            params["elgfaktor"] - number_steps * params["elg_perc_correct"],
+            params["perc_deduct_limit"],
+        )
+
+    else:
+        percentage = params["elgfaktor"]
+
+    return percentage
+
+
+def calc_net_wage(person):
+    """ Calculating the net wage of any person given taxes and social security
+    contributions.
+
+    """
+    net_wage = (
+        person["m_wage"] - person["incometax"] - person["soli"] - person["svbeit"]
+    )
+
+    return net_wage
+
+
+def calc_geschw_bonus(elterngeld_calc, params):
+    """ Calculating the bonus for siblings.
+
+
+    According to § 2a parents of siblings get a bonus.
+    """
+    bonus_calc = params["elg_geschw_bonus_share"] * elterngeld_calc
+    bonus = max(
+        min(bonus_calc, params["elg_geschw_bonus_max"]), params["elg_geschw_bonus_min"]
+    )
+    return bonus
+
+
+def check_eligibilities(household, params):
+    """Check if a parent is eligible for elterngeld. If so, then also check if it is
+    eligible for multiple or sibling bonus.
+
+    """
+    household["elternzeit_anspruch"] = False
+    household["geschw_bonus"] = False
+    household["num_mehrlinge"] = 0
+
+    children = household[household["child"]]
+    # Are there any children
+    if len(children) > 0:
+        youngest_child = children[
+            (children["byear"] == np.max(children["byear"]))
+            & (children["bmonth"] == np.max(children["bmonth"]))
+        ]
+        # Get the birthdate of the youngest child(If "Mehrlinge", then just take one)
+        birth_date = datetime.date(
+            day=youngest_child["bday"].iloc[0].astype(int),
+            month=youngest_child["bmonth"].iloc[0].astype(int),
+            year=youngest_child["byear"].iloc[0].astype(int),
+        )
+        age_youngest_child = relativedelta.relativedelta(params["date"], birth_date)
```

Comment:
what if this is `< 0`? Set it to zero or raise Exception? I vote for the latter.

Reply:
This would mean the birthdate of the child is after the date for the desired policy. Will we allow to calculate taxes for individuals, which have not lived at the policy date? If no, then this should be a general check in the beginning of the gettsim code.

Reply:
If someone runs own households with this, there is most likely a mishap concerning the correct dates. At the very least, we should issue a warning.

Reply:
Raised Error!